cpu/saml21/cpu.c

```diff
@@ -292,8 +296,8 @@ void cpu_init(void)
         }
     }
     /* clock used by timers */
-    _gclk_setup(SAM0_GCLK_8MHZ, GCLK_GENCTRL_GENEN | GCLK_GENCTRL_SRC_OSC16M
-                | GCLK_GENCTRL_DIV(2));
+    _gclk_setup(SAM0_GCLK_TIMER, GCLK_GENCTRL_GENEN | GCLK_GENCTRL_SRC_OSC16M
+                | GCLK_GENCTRL_DIV(CLOCK_CORECLOCK/sam0_gclk_freq(SAM0_GCLK_TIMER)));
```

Comment:
If CLOCK_CORECLOCK is set 48MHz, we will get a wrong frequency here.

Reply:
How did this work before? 48 MHz / 2 doesn't yield 8 MHz

Reply:
yea this should depend on the frequency of `OSC16M`, not `CLOCK_CORECLOCK`

Reply:
Yes, you are perfectly right! Thanks for spotting the bug, I'll correct it immediately.

Reply:
I've patched it in a way that works as it used to: but on second thought, it might be possibile to avoid OSC16M entirely. In case DFLL48M is active, we could source SAM0_GCLK_TIMER from that. What do you think?

Reply:
Something like this seems sensible - no need to activate a second oscillator at all:

```
#if USE_DFLL
#define GCLK_GENCTRL_SRC_MAIN GCLK_GENCTRL_SRC_DFLL48M
#else
#define GCLK_GENCTRL_SRC_MAIN GCLK_GENCTRL_SRC_OSC16M
#endif
    _gclk_setup(SAM0_GCLK_MAIN, GCLK_GENCTRL_GENEN | GCLK_GENCTRL_SRC_MAIN);

...

    /* clock used by timers */
    _gclk_setup(SAM0_GCLK_TIMER, GCLK_GENCTRL_GENEN | GCLK_GENCTRL_SRC_MAIN
                | GCLK_GENCTRL_DIV(CLOCK_CORECLOCK/sam0_gclk_freq(SAM0_GCLK_TIMER)));
```

Reply:
Of course this is desirable but I think this will can belong to another PR to keep things clear.

Reply:
I've measured power usage, the gain from this last solution is quite negligible, at about 0.1mA (2.52mA with timer clock fed from OSC16, 2.42mA with timer fed from DFLL48M).

Looks cleaner to me, though.

Reply:
> Of course this is desirable but I think this will can belong to another PR to keep things clear.

Ooops, sorry, missed your response and commited. Shall I revert and open a separate PR?

Reply:
> Ooops, sorry, missed your response and commited. Shall I revert and open a separate PR?

This is fine, I'll deal it with that way.